Sources/SwiftGRPC/Core/Channel.swift

```diff
@@ -14,28 +14,26 @@
  * limitations under the License.
  */
 #if SWIFT_PACKAGE
-  import CgRPC
-  import Dispatch
+import CgRPC
+import Dispatch
 #endif
 import Foundation
 
 /// A gRPC Channel
 public class Channel {
   /// Pointer to underlying C representation
   private let underlyingChannel: UnsafeMutableRawPointer
-
   /// Completion queue for channel call operations
   private let completionQueue: CompletionQueue
+  /// Observer for connectivity state changes.
+  private lazy var connectivityObserver = ConnectivityObserver(underlyingChannel: self.underlyingChannel)
```

Comment:
I'm concerned about making this lazy; this means we would always spin up an observer in `deinit`. Should we make this an optional and populate the variable in `addConnectivityObserver` if it is still nil? While at it, should we synchronize access to this variable? (Depends on whether we want to support adding connectivity observers from different threads.)

Reply:
Ahh yes great catch. I'll definitely update